src/provider_manifests.py

```diff
@@ -68,37 +63,24 @@ class UpdateNode(Patch):
     """Update the node manager daemonset as a patch."""
 
     NAME = "cloud-node-manager"
-    REQUIRED = {
-        "control-node-selector",
-    }
-
-    def _adjuster(self, toleration: Toleration) -> List[Toleration]:
-        node_selector = self.manifests.config.get("control-node-selector", {})
-        if toleration.key and toleration.key.startswith("node-role.kubernetes.io"):
-            return [
-                Toleration(
-                    key=key,
-                    value=value,
-                    effect=toleration.effect,
-                    operator=toleration.operator,
-                    tolerationSeconds=toleration.tolerationSeconds,
-                )
-                for key, value in node_selector.items()
-            ]
-        return [toleration]
 
     def __call__(self, obj):
         """Update the DaemonSet object in the cloud-node-manager."""
         if not (obj.kind == "DaemonSet" and obj.metadata.name == self.NAME):
             return
-        node_selector = self.manifests.config.get("control-node-selector")
-        if not isinstance(node_selector, dict):
-            log.error(
-                f"provider control-node-selector was an unexpected type: {type(node_selector)}"
-            )
-            return
 
-        update_toleration(obj, self._adjuster)
+        current_keys = {toleration.key for toleration in obj.spec.template.spec.tolerations}
+        missing_tolerations = [
+            Toleration(
+                key=taint.key,
+                value=taint.value,
+                effect=taint.effect,
+            )
+            for taint in self.manifests.config.get("control-node-taints", [])
+            if taint.key not in current_keys
```

Comment:
could there be a case where a key currently defined in the spec has the same key but a different value/effect in the kube-control relation data?

Reply:
Yes, i really believe there could be.  I'm generally frustrated at the fact that tolerations are a `list` built of key, value, operation.... etc. 

I need a a transform that does this: 
`    list -> set,  + missing,  -> list  `  where the "uniquifer" in the set is which fields are tolerated

I think this will be an adjustment to `ops.manifest` to just do tolerations better.  The way i've done now is suboptimal.

Tolerations are just that -- a list of things a POD can tolerate. There's no harm in tolerating the same set of criteria more than once -- but it's would just be an optimized list.  The important thing is to have the expected tolerations in the list on THIS resource to make sure it tolerates the controller's taints.